File: packages/shared/src/types/website.ts
import { z } from 'zod'

export const Website = z.object({
  id: z.string(),
  name: z.string(),
  domain: z.string(),
  widgetConfig: z.object({
    primaryColor: z.string().default('#3b82f6'),
    position: z.enum(['bottom-right', 'bottom-left']).default('bottom-right'),
    welcomeMessage: z.string().default('Hi! How can we help you?'),
    offlineMessage: z.string().default('We are currently offline. Leave us a message!'),
    enableFileUpload: z.boolean().default(true),
    enableEmoji: z.boolean().default(true),
  }),
  isActive: z.boolean().default(true),
  createdAt: z.date(),
  updatedAt: z.date(),
})

export type Website = z.infer<typeof Website>

export const CreateWebsiteRequest = Website.omit({
  id: true,
  createdAt: true,
  updatedAt: true,
})

export type CreateWebsiteRequest = z.infer<typeof CreateWebsiteRequest>

export const UpdateWebsiteRequest = Website.partial().omit({
  id: true,
  createdAt: true,
  updatedAt: true,
})

export type UpdateWebsiteRequest = z.infer<typeof UpdateWebsiteRequest>

File: apps/backend/app/models/website.py
from sqlalchemy import Boolean, Column, DateTime, String, JSON
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.database import Base
from app.models.user import user_website_association

class Website(Base):
    __tablename__ = "websites"

    id = Column(String, primary_key=True, index=True)
    name = Column(String, nullable=False)
    domain = Column(String, nullable=False, index=True)
    widget_config = Column(JSON, nullable=False, default={
        "primaryColor": "#6366f1",
        "position": "bottom-right",
        "welcomeMessage": "Hi! How can we help you today?",
        "agentName": "Support Team",
        "agentAvatar": None,
        "enableFileUpload": True,
        "enableEmoji": True,
        "offlineMessage": "We are currently offline. Leave us a message!"
    })
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # Relationships
    users = relationship("User", secondary=user_website_association, back_populates="websites")
    visitors = relationship("Visitor", back_populates="website")
    conversations = relationship("Conversation", back_populates="website")